libs/x-content/src/main/java/org/elasticsearch/common/xcontent/MediaTypeParser.java

```diff
@@ -86,7 +76,12 @@ public ParsedMediaType parseMediaType(String headerValue) {
                         if (keyValueParam.length != 2 || hasSpaces(keyValueParam[0]) || hasSpaces(keyValueParam[1])) {
                             return null;
                         }
-                        parameters.put(keyValueParam[0].toLowerCase(Locale.ROOT), keyValueParam[1].toLowerCase(Locale.ROOT));
+                        String parameterName = keyValueParam[0].toLowerCase(Locale.ROOT);
+                        String parameterValue = keyValueParam[1].toLowerCase(Locale.ROOT);
+                        if(isValidParameter(typeWithSubtype,parameterName,parameterValue) == false) {
+                            return null;
```

Comment:
should we throw an exception here? Maybe one that returns a 400 since the parameter is invalid

Reply:
but should it only be here? or should we throw exceptions in other places where we return null.
I think we should throw an exception when a media type does not conform to rfc
but if we don't know about the requested media type I think it might cause problems in code at the moment (we rely on checking if parsing returned null and then try a different method. See https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/AbstractRestChannel.java#L100

Reply:
as mentioned here this PR will not introduce throwing exceptions yet. I will follow up in another PR

Reply:
Can you log an issue for this ?

Reply:
done - https://github.com/elastic/elasticsearch/issues/63080 will try to gather more details before starting work on it